src/tests/Microsoft.PowerFx.Interpreter.Tests/BlobTestFunctions.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    public static class BlobTestFunctionsExtensions
+    {
+        public static void AddBlobTestFunctions(this PowerFxConfig config)
+        {
+            config.AddFunction(new BlobFunctionImpl());            
+            config.AddFunction(new BlobGetStringFunctionImpl());
+            config.AddFunction(new BlobGetBase64StringFunctionImpl());
+        }
+    }
+
+    internal class BlobFunction : BuiltinFunction
+    {
+        public BlobFunction()
+             : base("Blob", (loc) => "Converts a string to a Blob.", FunctionCategories.Text, DType.Blob, 0, 1, 2, DType.String, DType.Boolean)
+        {
+        }
+
+        public override bool IsSelfContained => true;
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new TexlStrings.StringGetter[] { (loc) => "string" };
+            yield return new TexlStrings.StringGetter[] { (loc) => "string", (loc) => "isBase64String" };            
+        }
+    }
+
+    internal class BlobFunctionImpl : BlobFunction, IAsyncTexlFunction5
+    {
+        public Task<FormulaValue> InvokeAsync(IServiceProvider runtimeServiceProvider, FormulaType irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();                       
+
+            if (args[0] is BlankValue)
+            {
+                return Task.FromResult<FormulaValue>(BlobValue.NewBlob(bytes: null));
+            }
```

Comment:
I know this is a test function, but we normally propagate blank values in constructor functions, so this could return blank value as well.